Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt

```diff
@@ -68,29 +67,32 @@ open class BpkButton(
       var paddingHorizontal = paddingHorizontal
       val paddingVertical = paddingVertical
 
-      if (iconPosition == ICON_ONLY) {
-        paddingHorizontal = resources.getDimensionPixelSize(R.dimen.bpkSpacingMd)
+      if (value == ICON_ONLY) {
+        paddingHorizontal = paddingHorizontalIconOnly
+        iconPadding = 0
       } else {
-        compoundDrawablePadding = resources.getDimensionPixelSize(R.dimen.bpkSpacingSm)
+        iconPadding = resources.getDimensionPixelSize(R.dimen.bpkSpacingMd)
       }
 
-      setPadding(paddingHorizontal, paddingVertical, paddingHorizontal, paddingVertical)
+      setPaddingRelative(paddingHorizontal, paddingVertical, paddingHorizontal, paddingVertical)
```

Comment:
Why did we switch here to padding relative?

Reply:
that's for good measure for RTL support - in this case it doesn't change anything as we've got the same padding on both sides, but it's good practise to use `relative`